wordpress-mobile/WordPress-Android
# [Blogging Prompts] Open Reader when tapping view all responses in card

Fixes #17735 
Fixes #17792

This PR implements a few things related to handling the "view responses" action:
- Change the answer text to say "View all responses"
- Open Reader searching for `dailyprompt-{prompt_id}` tag
- Track the analytics event `blogging_prompts_my_site_card_view_answers_tapped` when "view answers" button is tapped
- Track the analytics event `reader_tag_previewed` with  properties: `{"tag":"dailyprompt-<id>", "source":"blogging_prompts_my_site_card_view_answers"}`

Also hiding all of those changes behind the Blogging Prompts Enhancements `FeatureConfig` (`BloggingPromptsEnhancementsFeatureConfig`), keeping the old behavior if the flag is off.

The accessibility of the "answers" section of the Prompts card was also updated to make more sense when read by screen readers.

Demo:

https://user-images.githubusercontent.com/5091503/214138390-08f181b4-0e2a-4aa3-8fe9-27f6612a585d.mp4

To test:
1. Install Jetpack and log in
2. Open "Debug settings" and make sure `BloggingPromptsEnhancementsFeatureConfig` is disabled ("Features in Development" section)
3. Open "My Site" -> "HOME" with a blog selected
4. **Verify** the Blogging Prompts card is shown
5. **Verify** the text close to the Avatars says "<number> answers" and is not clickable
6. Open "Debug settings" and enable `BloggingPromptsEnhancementsFeatureConfig` (click "Restart the app" to apply)
7. Open "My Site" -> "HOME" with a blog selected
8. **Verify** the Blogging Prompts card is shown
9. **Verify** the text close to the Avatars says "View all responses"
10. Tap either the avatars or the "View all responses" text
11. **Verify** event tracked: `blogging_prompts_my_site_card_view_answers_tapped`
12. **Verify** the Reader is opened for tag `dailyprompt-<prompt-id>`
13. **Verify** event tracked: `reader_tag_previewed, Properties: {"tag":"dailyprompt-<prompt-id>", "source":"blogging_prompts_my_site_card_view_answers"}`

**Important**: currently there's a caveat in testing this feature, the `dailyprompt-<prompt-id>` tag is still not being set for prompt answers, so the Reader will most likely show an error / empty screen, so rely on the event tracked in step 13 above ( `reader_tag_previewed`) to check which tag was passed. This will only be able to be completely tested when the Blogging Prompts enhancements are released (or when web makes the changes to use the tag).

## Regression Notes
1. Potential unintended areas of impact
Showing updated enhanced behavior / features when the Feature Flag is off.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and updated unit tests.

3. What automated tests I added (or what prevented me from doing so)
Several tests were updated and added to test: feature toggle is working, tracking, correct messages displayed, proper tap handling.

PR submission checklist:
